src/torrent_handle.cpp

```diff
@@ -361,6 +361,11 @@ namespace libtorrent
 		TORRENT_ASYNC_CALL(queue_down);
 	}
 
+	void torrent_handle::queue_position_set(int p) const
+	{
+		TORRENT_ASYNC_CALL1(set_queue_position, p < 0 ? 0 : p);
```

Comment:
you should have a ``TORRENT_ASSERT_PRECOND(p >= 0, "queue position must be positive");`` here